sawmill-core/src/test/java/io/logz/sawmill/processors/UrlDecodeProcessorTest.java

```diff
@@ -0,0 +1,86 @@
+package io.logz.sawmill.processors;
+
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.utilities.JsonUtils;
+import org.junit.Test;
+
+import java.net.URLDecoder;
+import java.util.List;
+import java.util.Map;
+
+import static io.logz.sawmill.utils.FactoryUtils.createConfig;
+import static io.logz.sawmill.utils.FactoryUtils.createProcessor;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class UrlDecodeProcessorTest {
+
+    private final String messageExample = "{\n" +
+            "  \"innerObject\": {\n" +
+            "    \"url\": \"https://github.com/logzio/sawmill/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed\",\n" +
+            "    \"anotherUrl\": \"https%3A%2F%2Fwww.google.com%2Fsearch%3Fq%3Durl%2Bencode%2Bexample%26oq%3Durl%2Bencode%2Bexample%26aqs%3Dchrome.0.0l6.2790j0j4%26sourceid%3Dchrome%26ie%3DUTF-8\",\n" +
+            "    \"friends\": [{\n" +
+            "        \"id\": \"1%20%3A\",\n" +
+            "        \"url\": \"https%3A%2F%2Fwww.google.com%2Fsearch%3Fq%3Durl%2Bencode%2Bexample%26oq%3Durl%2Bencode%2Bexample%26aqs%3Dchrome.0.0l6.2790j0j4%26sourceid%3Dchrome%26ie%3DUTF-8\"\n" +
+            "      },\n" +
+            "      {\n" +
+            "        \"id\": \"%%1%20%3A\",\n" +
+            "        \"url\": \"https%3A%2F%2Fwww.google.com%2Fsearch%3Fq%3Durl%2Bencode%2Bexample%26oq%3Durl%2Bencode%2Bexample%26aqs%3Dchrome.0.0l6.2790j0j4%26sourceid%3Dchrome%26ie%3DUTF-8\"\n" +
+            "      },\n" +
+            "      {\n" +
+            "        \"id\": \"1%20%3A\",\n" +
+            "        \"url\": \"https%3A%2F%2Fwww.google.com%2Fsearch%3Fq%3Durl%2Bencode%2Bexample%26oq%3Durl%2Bencode%2Bexample%26aqs%3Dchrome.0.0l6.2790j0j4%26sourceid%3Dchrome%26ie%3DUTF-8\"\n" +
+            "      }\n" +
+            "    ]\n" +
+            "  },\n" +
+            "\"url\": \"https://github.com/logzio/sawmill/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed\",\n" +
+            "\"id\": \"1%20%3A\"\n" +
+            "}";
+
+
```

Comment:
test failures
maybe %%
and doc without field

Reply:
test already have %% , but i didnt add a test without a field i will add it.